cves/CVE-2017-12616.yml

```diff
@@ -171,8 +181,10 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      should have used whitelisting for request inputs, that way the requests
+      used for the vulnerability would have been rejected.
   security_by_obscurity:
```

Comment:
As far as lessons go, could the specially crafted requests be an example of complex inputs?